File: 0001_0100/0057/0057.py
def dist(a, b):
    return ((a[0] - b[0]) ** 2 + (a[1] - b[1]) ** 2) ** 0.5
 
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, c, p = (int(x) for x in f.readline().split())
    dots = []
    for i in range(n + 1):
        dots.append([int(x) for x in f.readline().split()])
    xs = [x[0] for x in dots]
    ys = [x[1] for x in dots]
    a = (sum(xs) / n, sum(ys) / n)
    m = (dots[0], dist(dots[0], a))
    for dot in dots[1:-1]:
        if dist(dot, a) < m[1]:
            m = (dot, dist(dot, a))
    q.write("YES" if (sum(dist(m[0], dot) for dot in dots) * c <= p) else "NO")

File: 0001_0100/0057/0057.go
package main

import (
	"fmt"
	"math"
	"os"
)

type dot struct {
	x, y int
}

func sum(a []float64) float64 {
	s := 0.0
	for _, v := range a {
		s += v
	}
	return s
}

func dist(a, b dot) float64 {
	return math.Sqrt(float64((a.x-b.x)*(a.x-b.x) + (a.y-b.y)*(a.y-b.y)))
}

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n int
	var c, p float64
	r := "NO"
	fmt.Fscan(f, &n, &c, &p)
	t := make([]dot, n+1)
	lens := make([][]float64, n+1)
	for i := 0; i <= n; i++ {
		fmt.Fscan(f, &t[i].x, &t[i].y)
		lens[i] = make([]float64, n+1)
	}
	for i := 1; i < n+1; i++ {
		for j := 0; j < i; j++ {
			lens[i][j] = dist(t[i], t[j])
			lens[j][i] = lens[i][j]
		}
	}
	for i := 0; i < n; i++ {
		if sum(lens[i])*c <= p {
			r = "YES"
			break
		}
	}
	fmt.Fprint(q, r)
}
